### PR TITLE
Redirects

### PR DESCRIPTION
**Pull request by trieloff**
Checks helix-redirect before serving content.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -82,8 +88,36 @@ async function executeActions(params) {
...
+
+    if (type === 'temporary' || type === 'permanent') {
+      log.info(`${type} redirect to ${target}`);
+      return sendRedirect(redirectPromise);
+    } else if (type === 'internal' && target) {
+      // increase the internal redirect counter
+      const redirects = (params.redirects || 0) + 1;
+      if (redirects > MAX_REDIRECTS) {
+        log.warn(`${type} redirect to ${target} exceeds redirect counter`);
+        return abortRedirect(redirectPromise);
```
Comment (tripodsan) [inline src/index.js:107]: same as above....I would just pass in the `result`

File: src/redirects.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      path: params.path,
+    },
+    blocking: true,
+    result: true,
+  };
+
+  const result = await ow.actions.invoke(opts);
+  return {
+    type: TYPES[result.statusCode] || null,
+    target: result.headers ? result.headers.Location : null,
```
[the remark this answers is not in the document]
Comment (trieloff) [reply src/redirects.js:34]: Yes, `helix-fetch` (unlike OW) does not downcase header names.

[the remark this answers is not in the document]
Comment (tripodsan) [reply src/redirects.js:34]: but in helix-fetch, the headers is a map. so you should use `result.headers.get('location')`. no ?

https://developer.mozilla.org/en-US/docs/Web/API/Headers

[the remark this answers is not in the document]
Comment (trieloff) [reply src/redirects.js:34]: Sorry, that answer was only 50% correct and 50% BS. Helix Fetch isn't used here. The header is uppercase and it won't get lowercased on the way out.

File: src/index.js [hunk 3]
```diff
@@ -82,8 +88,36 @@ async function executeActions(params) {
...
+    const redirectPromise = redirect(params, ow);
+
+    // start the fetching processes
+    const responsePromise = resolvePreferred(fetchers(params, log).map(invoker));
+
+    const { type, target } = await redirectPromise;
+
+    if (type === 'temporary' || type === 'permanent') {
+      log.info(`${type} redirect to ${target}`);
+      return sendRedirect(redirectPromise);
```
[the remark this answers is not in the document]
Comment (trieloff) [reply src/index.js]: ok

File: src/redirects.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    result,
+  };
+}
+
+async function sendRedirect(redirectPromise) {
```
[the remark this answers is not in the document]
Comment (trieloff) [reply src/redirects.js]: ok.

File: src/index.js (hunk 1 above)
Comment (trieloff) [reply src/index.js]: This is a bit different, as the only thing needed is the target, but the response is entirely new.

**Pull request merged**